pkgs/development/interpreters/python/cpython/2.7/default.nix

```diff
@@ -60,6 +60,24 @@ let
 
       ./properly-detect-curses.patch
 
+      # Fix python bug #27177 (https://bugs.python.org/issue27177)
+      # The issue is that `match.group` only recognizes python integers
+      # instead of everything taht has `__index__`.
+      # This bug was fixed upstream, but not backported to 2.7
+      (fetchpatch {
+        name = "re_match_index.patch";
+        url = "https://bugs.python.org/file43084/re_match_index.patch";
+        sha256 = "0l9rw6r5r90iybdkp3hhl2pf0h0s1izc68h5d3ywrm92pq32wz57";
+      })
+
+      # "`type_getattro()` calls `tp_descr_get(self, obj, type)` without actually owning a reference to "self".
+      # In very rare cases, this can cause a segmentation fault if "self" is deleted by the descriptor."
+      # https://github.com/python/cpython/pull/6118
```

Comment:
Similarly, reading that PR suggests this should be applied to all python versions (although it does say backporting to 2.7 is the most important since it's most problematic there).

Reply:
Thats true, I also only applied this to 2.7 to minimize rebuilds. But since this is a crashing bug, here it might really be a good idea to add it to all versions. Especially since the other PR triggers rebuilds across all versions anyways -- but that only matters if this gets merged before those builds run (which is a bit optimistic).

Reply:
Minimizing mass rebuilds isn't really needed, I believe.  On staging we have a large rebuild almost every week, and python2 by itself is probably the worst part.